File: ds_easily_written_in_python/Go/myutils/checkbrackets.go
package myutils

import (
	slicestack "../dataStruct/SliceStack"
)

func CheckBrackets(statement string) bool {
	stack := slicestack.NewSliceStack()
	for _, ch := range statement {
		if ch == '{' || ch == '[' || ch == '(' {
			stack.Push(ch)
		} else if ch == '}' || ch == ']' || ch == ')' {
			if stack.IsEmpty() {
				return false
			} else {
				left := stack.Pop()
				if (ch == '}' && left != '{') || (ch == ']' && left != '[') || (ch == ')' && left != '(') {
					return false
				}
			}
		}
	}
	return stack.IsEmpty()
}

func CheckBracketsV2(statement string) bool {
	stack := slicestack.NewSliceStack()
	for ch := range statement {
		if ch == '{' || ch == '[' || ch == '(' {
			stack.Push(ch)
		} else if ch == '}' || ch == ']' || ch == ')' {
			if stack.IsEmpty() {
				return false
			} else {
				left := stack.Pop()
				if (ch == '}' && left != '{') || (ch == ']' && left != '[') || (ch == ')' && left != '(') {
					return false
				}
			}
		}
	}
	return stack.IsEmpty()
}

File: ds_easily_written_in_python/python/myutil/checkBrackets.py
import os
import sys
sys.path.append(os.path.dirname(os.path.abspath(os.path.dirname(__file__))))

from dataStruct import ArrayStack

def checkBrackets(statement):
    stack = ArrayStack.ArrayStack()
    for ch in statement:
        if ch in ('{[('):
            stack.push(ch)
        elif ch in ('}])'):
            if stack.isEmpty():
                return False
            else:
                left = stack.pop()
                if (ch == "}" and left != "{") or \
                    (ch == "]" and left != "[") or \
                    (ch == ")" and left != "("):
                    return False
    return stack.isEmpty()

def checkBracketsV2(lines):
    stack = ArrayStack.ArrayStack()
    for line in lines:
        for ch in line:
            if ch in ('{[('):
                stack.push(ch)
            elif ch in ('}])'):
                if stack.isEmpty():
                    return False
                else:
                    left = stack.pop()
                    if (ch == "}" and left != "{") or \
                        (ch == "]" and left != "[") or \
                        (ch == ")" and left != "("):
                        return False
    return stack.isEmpty()
 

